node/core/pvf/src/host.rs

```diff
@@ -446,11 +457,28 @@ async fn handle_precheck_pvf(
 				*last_time_needed = SystemTime::now();
 				let _ = result_sender.send(Ok(()));
 			},
-			ArtifactState::Preparing { waiting_for_response } =>
+			ArtifactState::Preparing { waiting_for_response, num_failures: _ } =>
 				waiting_for_response.push(result_sender),
-			ArtifactState::FailedToProcess(result) => {
-				let _ = result_sender.send(PrepareResult::Err(result.clone()));
-			},
+			ArtifactState::FailedToProcess { last_time_failed, num_failures, error } =>
+				if can_retry_prepare_after_failure(*last_time_failed, *num_failures) {
```

Comment:
NIT: For preparation we actually agreed that retrying is not required/desired here. The pre-checking logic will not actually retry anyway, so this code will never get executed - so it is fine, but given that not doing it actually simplifies things, instead of complicating things (due to this complete separate implementation) - I would suggest to just not do it.

Reply:
Would also remove some code duplication.

Reply:
Makes sense, removed.